### PR TITLE
Update and improve interactivity of demographic plot

### LINKED ISSUE #45: Implement the cards for Percentage of deaths by Demographics

### PR DESCRIPTION
**Pull request by billwan96**
closes #45

### CODE REVIEW

**Comment by billwan96** [comment]
*Note: some instruction should be added since filter of gender and youth would not work in this plot
(related to #51 )

File: src/app.py [hunk 1]
```diff
@@ -107,31 +107,34 @@
...
      Input('year_range_slider', 'value')]
-     )
+)
 def update_figure(selected_drug, selected_years):
     if selected_drug == ['Overall']:
         selected_drug = ['Total Overdose Deaths']
+        title = "Overall Overdose Death Rate based on Demographic"
     else:
         selected_drug = selected_drug
+        title = f"Overdose Death Rate based on Demographic <br>for {' and '.join(selected_drug[1:])}"
```
Comment (SoloSynth1) [inline src/app.py:118]: I think this would make the title unnecessarily cluttered when the user select 3+ items in the checklist.

**Review by SoloSynth1 — COMMENTED** [review]
@billwan96 Please see comment. 

On the other hand, now that you're working on the drug filter, can you also change the default values of the checklist, such that all items would be selected by default? I think this would make more sense than with all of them unselected by default.

**Comment by billwan96** [comment]
this is a good catch thanks @SoloSynth1.
However, it comes back to the problem that just adding the rate of each drug type, will not be the same as the total...
Do you guys think it is alright (we just add a note below the plot?), or any other thoughts @jinyz8888 @AlysenTownsley

**Comment by SoloSynth1** [comment]
Yeah, double counting across drug types is a known issue. Agreed that adding a footnote would make users informed, but I don't think that is a priority right now. Maybe we can create an issue for next Milestone to address this.

**Review by AlysenTownsley — COMMENTED** [review]
I think the plot title should be in a smaller font (could be an enhancement for next week), as it currently runs off of the page.

**Comment by billwan96** [comment]
> I think the plot title should be in a smaller font (could be an enhancement for next week), as it currently runs off of the page.

It is a bit difficult since the selection is changing the length of the title... but I agree it isn't very nice right now

**Comment by SoloSynth1** [comment]
> I think the plot title should be in a smaller font (could be an enhancement for next week), as it currently runs off of the page.
 
Can be changed it via CSS stylesheets, won't be too hard

**Comment by jinyz8888** [comment]
I agree with @SoloSynth1. We complete the key parts of this milestone as priority, and then optimize it later.

File: src/app.py (hunk 1 above)
Comment (SoloSynth1) [reply src/app.py]: Modified this behaviour in the latest commit such that now there is a HTML element capturing the change instead. Resolving this convo.

**Review by SoloSynth1 — APPROVED** [review]
LGTM. Thanks for implementing the filter and the callback @billwan96.

**Comment by SoloSynth1** [comment]
Merging into `main`.

**Pull request merged**